rpc/query_server.go

```diff
@@ -749,36 +756,44 @@ func (s *QueryServer) EthGetTransactionReceipt(hash eth.Data) (resp eth.JsonTxRe
 	if err != nil {
 		return resp, err
 	}
+
 	txReceipt, err := r.GetReceipt(snapshot, txHash)
 	if err != nil && errors.Cause(err) != common.ErrTxReceiptNotFound {
-		return resp, err
+		// return empty response if cannot find hash
+		resp.Status = eth.EncInt(int64(StatusTxFail))
+		return resp, nil
```

Comment:
Looks like `LevelDbReceipts.GetReceipt` won't ever return `common.ErrTxReceiptNotFound`, it'll return `ErrNotFound` instead, so it doesn't looks like `getReceiptByTendermintHash` will ever get called below.

Reply:
I should look for a way of making less confusing, but the code holds together.
 `r.GetReceipt(snapshot, txHash)` looks for the hash both in the leveldb and the chain.
```go
	case ReceiptHandlerLevelDb:
		receipt, err := r.leveldbReceipts.GetReceipt(txHash)
		// In case receipts has been upgraded from V1 to V2 try getting the receipt from the chain.
		if err != nil {
			var chainErr error
			receipt, chainErr = r.chainReceipts.GetReceipt(state, txHash)
			if chainErr != nil {
				return receipt, errors.Wrap(chainErr, err.Error())
			}
		}
		return receipt, nil
```
`levledbReceipts.GetReceipt` tends to return `leveldb: not found`, while `r.chainReceipts.GetReceipt` returns `common.ErrTxReceiptNotFound`. Its the `r.chainReceipts.GetReceipt` error that ends up being the cause.

Reply:
Changed it so it's more explicit that GetReceipt returns `common.ErrTxReceiptNotFound`.